test/types/records/generic/genericRecordInFunction.good

```diff
@@ -0,0 +1 @@
+genericRecordInFunction.chpl:3: error: could not determine the concrete type for the generic return type '<any type>'
```

Comment:
This doesn't seem like a very friendly user-facing error.  Where in the program is there a return type?

Should the program compile?

Reply:
The reference to the return type is coming from compiler generated wrappers around fields. It could have slightly more specific errors to this case. To my mind, the mention of needing a concrete type here in addition to correct line information makes this a decent error message.

Reply:
I agree with Paul.  I think it's reasonable to say that this program has improved, in that I do think it shouldn't encounter an internal compiler error, but I think it would be great for it to both:
- not mention something that isn't in the user's control
- probably actually compile, since R is fully contained within the contents of the function and so should be able to be concretized whenever foo gets instantiated

With that in mind, I might be inclined to adjust the issue to be an "Error message" and/or "Design" issue.  But I wouldn't object to closing it and opening a separate issue to track the next steps for it

Reply:
I will keep the test as-is (with your additions per your other comment), but I will add a comment to the issue that this should either have a better error message or be allowed.